src/Illuminate/Database/Concerns/ManagesTransactions.php

```diff
@@ -65,6 +59,12 @@ public function transaction(Closure $callback, $attempts = 1)
                 continue;
             }
 
+            $this->transactionsManager?->commit(
+                $this->getName(),
+                $levelBeingCommitted,
```

Comment:
No guarantee this variable has even been defined at this point.

Reply:
There is because of the `continue;` statement in the `catch` block. 

It's either no exception (in which case the variable is defined), or an exception which is caught by the `catch` block and the `transactionsManager?->commit()` is not executed at all.

But I can move `$levelBeingCommitted` out of the `try-catch` if you wish.